packages/cdktf-cli/lib/checkpoint.ts

```diff
@@ -94,7 +110,19 @@ export async function ReportRequest(reportParams: ReportParams): Promise<void> {
     reportParams.os = os.platform();
   }
 
+  const ci: string | false = ciDetect();
+  if (!reportParams.userId && !ci) {
+    reportParams.userId = machineIdSync();
```

Comment:
In RFC review @eastebry suggested that we should add a static salt to this. I propose we use the string "cdktf-static-salt" or similar to be explicit that this is intentionally a static salt. The idea is that in case our checkpoint data were ever compromised, the hashes couldn't readily be matched to other datasets.